components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/profiler/Profiler.kt

```diff
@@ -63,4 +64,21 @@ class Profiler(
     override fun addMarker(markerName: String) {
         runtime.profilerController.addMarker(markerName)
     }
+
+    override fun startProfiler(filters: Array<String>, features: Array<String>) {
+        runtime.profilerController.startProfiler(filters,features)
+    }
+
+    override fun stopProfiler(onSuccess: (ByteArray?) -> Unit, onError: (Throwable) -> Unit) {
```

Comment:
todo (non-blocking): according [to the current code](https://searchfox.org/mozilla-central/rev/2fd70a667e6f89a9ec6622ecb302b1ab48e4a062/tools/profiler/core/platform.cpp#284), the `ByteArray` in `onSuccess` will never be null. We should enforce this here, in android-components, to simplify the code for fenix.